google/cloud/storage/blob.py

```diff
@@ -3426,6 +3426,13 @@ def open(
         This method can be used as a context manager, just like Python's
         built-in 'open()' function.
 
+        While reading, as with other read methods, if blob.generation is not set
+        the most recent blob generation will be used. Because the file-like IO
+        reader downloads progressively in chunks, this could result in data from
+        multiple versions being mixed together. If this is a concern, use
+        either bucket.get_blob(), or blob.reload(), which will download the
+        latest generation number and set it.
```

Comment:
Suggestion: Alternatively pass in the value when calling `bucket.blob(generation=123456)` which does not perform a request if the generation is available locally.